openeogeotrellis/deploy/run_graph_locally.py

```diff
@@ -1,33 +1,49 @@
 import os
+import sys
 from pathlib import Path
 
 import requests
+
+import openeogeotrellis.deploy.local
+from openeo.internal.graph_building import as_flat_graph
 from openeo.util import ensure_dir
-from openeo_driver.utils import read_json
+
+openeogeotrellis.deploy.local.setup_environment()
+
+# Can only import after setup_environment:
 from openeogeotrellis.deploy.batch_job import run_job
 
 # Avoid IPv6, to avoid hanging on https://services.terrascope.be/catalogue//collections
 requests.packages.urllib3.util.connection.HAS_IPV6 = False
 
-import openeogeotrellis.deploy.local
 
-openeogeotrellis.deploy.local.setup_local_spark(["/opt/"])  # TODO: cleanup /opt contains jars in the docker file
-
-workdir = Path(os.getcwd())
-
-process_graph_path = workdir / "process_graph.json"
-print("process_graph_path: " + str(process_graph_path))
-process_graph = read_json(process_graph_path)
-if "process_graph" not in process_graph and "job_options" not in process_graph:
-    print("Wrapping process graph")
-    process_graph = {"process_graph": process_graph}
-
-run_job(
-    process_graph,
-    output_file=workdir / "random_folder_name",
-    metadata_file=workdir / "metadata.json",
-    api_version="1.0.0",
-    job_dir=ensure_dir(workdir / "job_dir"),
-    dependencies=[],
-    user_id="jenkins",
-)
+def run_graph_locally(process_graph, output_dir):
+    output_dir = Path(output_dir)
+    process_graph = as_flat_graph(process_graph)
+    if "process_graph" not in process_graph:
+        process_graph = {"process_graph": process_graph}
+    run_job(
+        process_graph,
+        output_file=output_dir / "random_folder_name",
+        metadata_file=output_dir / "metadata.json",
+        api_version="1.0.0",
+        job_dir=ensure_dir(output_dir),
+        dependencies=[],
+        user_id="jenkins",
+    )
+
+
+def start_entrypoint():
+    """
+    for setup.py entry_points
+    """
+    if len(sys.argv) >= 2:
+        workdir = Path(sys.argv[1])
+    else:
+        workdir = Path.cwd()
+    process_graph_path = workdir / "process_graph.json"
```

Comment:
Do I understand correctly that the input JSON file must be called "process_graph.json", otherwise it won't work?

Reply:
True, make the name of the process graph file an argument